partiql-lang/src/main/kotlin/org/partiql/lang/eval/CompileOptions.kt

```diff
@@ -79,10 +79,6 @@ enum class TypingMode {
  * Indicates how CAST should behave.
  */
 enum class TypedOpBehavior {
-    @Deprecated(message = "TypedOpBehavior.LEGACY is an old compile option that will be removed in an upcoming release", replaceWith = ReplaceWith("TypedOpBehavior.HONOR_PARAMETERS"))
-    /** The old behavior that ignores type arguments in CAST and IS. */
-    LEGACY,
-
```

Comment:
With `HONOR_PARAMETERS` as the only value remaining in the enum, did we overlook the opportunity to deprecate all of `TypedOpBehavior`, as opposed to just its `LEGACY` value?

Could we be a little daring and remove `TypedOpBehavior` in this version cycle as well, skipping the deprecation ceremony? (This would be a separate PR.) 
From what I see, the primary concern would be various configuration methods that take a `TypedOpBehavior` argument.  But most (all?) of those default the argument to `HONOR_PARAMETERS` already -- that is, the argument can be skipped, and chances are is being skipped by users.  While the other value for the argument is already impossible. In other words, anyone who would bump into an upgrade hiccup with disappeared `TypedOpBehavior` would have bumped into a similar hiccup with disappeared `LEGACY`.

Reply:
Let's absolutely do that. I can look into it after this PR.